generators/generator-microsoft-bot-adaptive/generators/app/templates/dotnet/webapp/Controllers/BotController.cs

```diff
@@ -1,34 +1,63 @@
-﻿using System.Threading.Tasks;
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
 using Microsoft.AspNetCore.Mvc;
 using Microsoft.Bot.Builder;
 using Microsoft.Bot.Builder.Integration.AspNet.Core;
+using Microsoft.Bot.Builder.Integration.Runtime.Settings;
 
 namespace <%= botName %>.Controllers
 {
     // This ASP Controller is created to handle a request. Dependency Injection will provide the Adapter and IBot
     // implementation at runtime. Multiple different IBot implementations running at different endpoints can be
     // achieved by specifying a more specific type for the bot constructor argument.
-    [Route("api/messages")]
     [ApiController]
     public class BotController : ControllerBase
     {
-        private readonly IBotFrameworkHttpAdapter _adapter;
-
+        private readonly Dictionary<string, IBotFrameworkHttpAdapter> _adapters = new Dictionary<string, IBotFrameworkHttpAdapter>();
         private readonly IBot _bot;
 
-        public BotController(IBotFrameworkHttpAdapter adapter, IBot bot)
+        public BotController(
+            IEnumerable<IBotFrameworkHttpAdapter> adapters, 
+            IEnumerable<AdapterSettings> adapterSettings, 
+            IBot bot)
         {
-            this._adapter = adapter;
-            this._bot = bot;
+            _bot = bot ?? throw new ArgumentNullException(nameof(bot));
+
+            foreach (var adapter in adapters ?? throw new ArgumentNullException(nameof(adapters)))
+            {
+                var settings = adapterSettings.FirstOrDefault(s => s.Enabled && s.Name == adapter.GetType().FullName);
+
+                if (settings != null)
+                {
+                    _adapters.Add(settings.Route, adapter);
+                }
+            }
         }
 
         [HttpPost]
         [HttpGet]
-        public async Task PostAsync()
+        [Route("api/{route}")]
+        public async Task PostAsync(string route)
         {
-            // Delegate the processing of the HTTP POST to the adapter.
-            // The adapter will invoke the bot.
-            await this._adapter.ProcessAsync(Request, Response, _bot).ConfigureAwait(false);
+            if (string.IsNullOrEmpty(route))
+            {
+                throw new ArgumentNullException(nameof(route));
+            }
+
+            IBotFrameworkHttpAdapter adapter;
+
+            if (_adapters.TryGetValue(route, out adapter))
```

Comment:
Same comments here on inline out variable declaration and removal of else statement/inversion of condition.